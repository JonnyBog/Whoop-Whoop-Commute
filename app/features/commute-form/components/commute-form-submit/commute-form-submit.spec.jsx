import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CommuteFormSubmit from 'features/commute-form/components/commute-form-submit/commute-form-submit';

describe('Features', () => {
  describe('Commute Form Submit Component', () => {
    it('should export', () => {
      expect(CommuteFormSubmit).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        isFetching: false,
        error: 'test error',
        children: 'submit'
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createRenderTree
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} component tree
     */
    function createRenderTree (updatedProps = props) {
      return renderer.create(
        <CommuteFormSubmit {...updatedProps} />
      ).toJSON();
    }

    /**
     * createShallowWrapper
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} shallow component
     */
    function createShallowWrapper (updatedProps = props) {
      return shallow(
        <CommuteFormSubmit {...updatedProps} />
      );
    }

    describe('Appearance', () => {
      it('should render', () => {
        const tree = createRenderTree();
        expect(tree).toMatchSnapshot();
      });

      it('should not show message when isFetching is false', () => {
        const wrapper = createShallowWrapper();

        expect(wrapper.find('commute-form-submit__FetchingMessage')).toHaveLength(0);
      });

      it('should show message when isFetching', () => {
        const updatedProps = Object.assign(
          {},
          props,
          {
            isFetching: true
          }
        );
        const wrapper = createShallowWrapper(updatedProps);

        expect(wrapper.find('commute-form-submit__FetchingMessage')).toHaveLength(1);
      });
    });
  });
});
