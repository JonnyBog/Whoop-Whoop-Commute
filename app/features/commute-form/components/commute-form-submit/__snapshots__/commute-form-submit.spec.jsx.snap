// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Commute Form Submit Component Appearance should render 1`] = `
<button
  className="commute-form-submit__Submit-ci59mw-0 eJkQdd"
  disabled="test error"
  type="submit"
>
  submit
</button>
`;
